lib/ansible/modules/windows/win_get_url.ps1

```diff
@@ -19,120 +20,186 @@
 # WANT_JSON
 # POWERSHELL_COMMON
 
+
+Function CheckModified-File($url, $dest, $headers, $credentials, $timeout, $use_proxy, $proxy) {
```

Comment:
Sigh. We have 3 possibilities here:

1. No proxy_url, use user/system proxy settings (default behaviour now)
2. Custom proxy_url provided
3. We absolutely don't want proxies to be used

If you look at **get_url** they have the `use_proxy` parameter for this. This is exactly what we needed.

So if you only would provide `$proxy` to our functions, we can only do 2 things:

- $null object (this means, absolutely use NO proxy)
- Proxy object (use this proxy/credentials)

There is no room to tell the function that it should use the current default behaviour, use the system/user proxy configuration. We could invent some other value for `$proxy` that would mean 'Use the system/user proxy config' but that does not make the code any cleaner. And in fact, that's why we have the `$use_proxy` variable, which is already set.

So I really don't think we should device something more complex and awkward, just to prevent not sending 2 variables to the function. We definitely cannot have a single `$webRequest.Proxy = $proxy` in both functions, because it doesn't work that way.